website/source/docs/drivers/external/singularity.html.md

```diff
@@ -0,0 +1,187 @@
+    config {
+      debug = "false"
+    }
+    ```
+
+* `command` - Singularity command action; can be `run`, `exec` or `test`.
+
+    ```hcl
+    config {
+      command = "run"
+    }
+    ```
+
+* `args` - (Optional) Singularity command action arguments, when trying to pass arguments to `run`, `exec` or `test`.
+Multiple args can be given by a comma separated list.
+
+    ```hcl
+    config {
+      args = [ "echo", "hello Cloud" ]
+    }
+    ```
+
+* [`binds`][bind] - (Optional) A user-bind path specification. This spec has the format `src[:dest[:opts]]`, where src and
+dest are outside and inside paths.  If dest is not given, it is set equal to src.
+Mount options ('opts') may be specified as 'ro' (read-only) or 'rw' (read/write, which
+is the default). Multiple bind paths can be given by a comma separated list.
+
+    ```hcl
+    config {
+      bind = [ "host/path:/container/path" ]
+    }
+    ```
+
+* [`overlay`][overlay] - (Optional) Singularity command action flag, to enable an overlayFS image for persistent data
+storage or as read-only layer of container.  Multiple overlay paths can be given by a comma separated list.
+
+    ```hcl
+    config {
+      overlay = [ "host/path/to/overlay" ]
+    }
+    ```
+
+* [`security`][security] - (Optional) Allows the root user to leverage security modules such as
+SELinux, AppArmor, and seccomp within your Singularity container.
+You can also change the UID and GID of the user within the container at runtime.
+
+    ```hcl
+    config {
+      security = [ "uid:1000 " ]
+    }
+    ```
+
+* `contain` - (Optional) Use minimal `/dev` and empty other directories (e.g. /tmp and $HOME) instead of sharing filesystems from your host.
+
+    ```hcl
+    config {
+      contain = "false"
+    }
+    ```
+
+* `workdir` - (Optional) Working directory to be used for `/tmp`, `/var/tmp` and $HOME (if -c/--contain was also used).
+
+    ```hcl
+    config {
+      workdir = "/path/to/folder"
+    }
+    ```
+
+* `pwd` - (Optional) Initial working directory for payload process inside the container.
+
+    ```hcl
+    config {
+      pwd = "/path/to/folder"
+    }
+    ```
+
+## Networking
+
+Currently the `Singularity` driver only supports host networking. For more detailed instructions on how to set up networking options, please refer to the `Singularity` user guides [singularity-network]
+
+## Client Requirements
+
+The `Singularity` driver requires the following:
+
+* 64-bit Linux host
+* The `linux_amd64` Nomad binary
+* The Singularity driver binary placed in the [plugin_dir][plugin_dir] directory.
+* [`Singularity`][Singularity] v3.1.1+ to be installed
+
+## Plugin Options<a id="plugin_options"></a>
+
+* `enabled` - The `Singularity` driver may be disabled on hosts by setting this option to `false` (defaults to `true`).
+
+* `Singularity_path` - The location in which all containers are stored (commonly defaults to `/var/lib/Singularity`). See [`Singularity-cache`][Singularity-cache] for more details.
```

Comment:
the config schema returned suggests that this is [`singularity_path`](https://github.com/sylabs/nomad-driver-singularity/blob/master/pkg/plugin/driver.go#L129), although it doesn't appear to be used.

Reply:
is not used yet, I plan to use it, but as for alpha-rc1 is not used. I am trying to follow the LXC and grow from there. will update the docs from S to s